File: java-api-injection/src/main/java/Java_Api_Injection/service/ImageService.java
package Java_Api_Injection.service;

import java.util.List;

import org.springframework.stereotype.Service;

import Java_Api_Injection.constant.InjectionConstant;
import Java_Api_Injection.dao.ImageDao;
import Java_Api_Injection.model.Image;

@Service
public class ImageService {
  private final boolean SQL_INJECTION_ACTIVE = InjectionConstant.SQL_INJECTION_ACTIVE;
  ImageDao imageDao = new ImageDao();

  public List<Image> getAll() {
    return imageDao.getAll();
  }

  public List<Image> getAllByUser(int userId) {
    return imageDao.getAllByUser(userId);
  }

  public Image get(int id) {
    return SQL_INJECTION_ACTIVE ? imageDao.get(id) : imageDao.getSecure(id);
  }

  public List<Image> search(int userId, String input) {
    return SQL_INJECTION_ACTIVE ? imageDao.search(userId, input) : imageDao.searchSecure(userId, input);
  }
}

File: node-api-injection/api/src/services/ImageService.js
const { SQL_INJECTION_ACTIVE } = require('../constants/InjectionConstant.js')
const imageModel = require('../models/ImageModel.js')

const getAll = async () => {
  return await imageModel.getAll()
}

const getAllByUser = async (userId) => {
  return await imageModel.getAllByUser(userId)
}

const get = async (id) => {
  return SQL_INJECTION_ACTIVE
    ? await imageModel.get(id)
    : await imageModel.getSecure(id)
}

const search = async (userId, input) => {
  return SQL_INJECTION_ACTIVE
    ? await imageModel.search(userId, input)
    : await imageModel.searchSecure(userId, input)
}

module.exports = {
  getAll,
  getAllByUser,
  get,
  search,
}
